django_auth_ldap/backend.py

```diff
@@ -78,6 +78,31 @@
 ldap_error = django.dispatch.Signal()
 
 
+_error_context_descriptions = {
+    "authenticate": "while authenticating",
+    "populate_user": "populating user info",
+    "get_group_permissions": "loading group permissions",
+    "search_for_user_dn": "looking up user",
+    "mirror_groups": "updating mirrored groups",
+}
+
+
+def _report_error(sender, context, user, request, exception):
+    description = _error_context_descriptions.get(context, "from unknown context")
+    logger.warning(
+        "Caught LDAPError %s: %s",
+        description,
+        pprint.pformat(exception)
+    )
+    ldap_error.send(
```

Comment:
Now that I think about this: the order of `logger.warning` and `ldap_error.send` is significant, and I changed the order.

If the app developer raises an exception from `ldap_error` signal handler, then previously no log message was emitted from django_auth_ldap itself: it was up to whoever handles the thrown exception to log the error.

I'm unsure which is better.

Reply:
I would always log (so log before sending the signal). Logging helps with auditing, and we don’t want to log the issue only when the signal handlers did not raise an exception.